superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -270,6 +327,22 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
               data-test="new-chart-name"
             />
           </FormItem>
+          {this.props.datasource && this.props.datasource.type === 'query' ? (
```

Comment:
this can be simplified using optional chaining:
this.props.datasource?.type === 'query'